css/geometry-1/DOMRect-002.html

```diff
@@ -0,0 +1,168 @@
+            assert_equals(r.bottom, 0, "Expected value for bottom is 0");
+        },'testReadOnlySetAttributes');
+
+        test(function() {
+            var actual = DOMRect.fromRect({x: 1, y: 2, width: 3, height: 4});
+            var expected = new DOMRect(1, 2, 3, 4);
+            checkDOMRect(actual, expected);
+        },'testFromRect');
+
+        test(function() {
+            var actual = DOMRectReadOnly.fromRect({x: 1, y: 2, width: 3, height: 4});
+            var expected = new DOMRectReadOnly(1, 2, 3, 4);
+            checkDOMRect(actual, expected);
+        },'testReadOnlyFromRect');
+
+        function testConstructor(constructor) {
+            var prefix = constructor === DOMRect ?  'testConstructor' : 'testReadOnlyConstructor';
+            test(function() {
+                checkDOMRect(new constructor(),
+                    { x: 0, y: 0, width: 0, height: 0, top: 0, right: 0, bottom: 0, left: 0 });
+            }, prefix + '0');
+            test(function() {
+                checkDOMRect(new constructor(1),
+                    { x: 1, y: 0, width: 0, height: 0, top: 0, right: 1, bottom: 0, left: 1 });
+            }, prefix + '1');
+            test(function() {
+                checkDOMRect(new constructor(1, 2),
+                    { x: 1, y: 2, width: 0, height: 0, top: 2, right: 1, bottom: 2, left: 1 });
+            }, prefix + '2');
+            test(function() {
+                checkDOMRect(new constructor(1, 2, 3),
+                    { x: 1, y: 2, width: 3, height: 0, top: 2, right: 4, bottom: 2, left: 1 });
+            }, prefix + '3');
+            test(function() {
+                checkDOMRect(new constructor(1, 2, 3, 4),
+                    { x: 1, y: 2, width: 3, height: 4, top: 2, right: 4, bottom: 6, left: 1 });
+            }, prefix + '4');
+            test(function() {
+                checkDOMRect(new constructor(1, 2, 3, 4, 5),
+                    { x: 1, y: 2, width: 3, height: 4, top: 2, right: 4, bottom: 6, left: 1 });
+            }, prefix + '5');
+            test(function() {
+                checkDOMRect(new constructor(2, 2, -4, 4),
+                    { x: 2, y: 2, width: -4, height: 4, top: 2, right: 2, bottom: 6, left: -2 });
+            }, prefix + 'NegativeWidth');
+            test(function() {
+                checkDOMRect(new constructor(2, 2, 4, -4),
+                    { x: 2, y: 2, width: 4, height: -4, top: -2, right: 6, bottom: 2, left: 2 });
+            }, prefix + 'NegativeHeight');
+            test(function() {
+                checkDOMRect(new constructor(2, 2, -4, -4),
+                    { x: 2, y: 2, width: -4, height: -4, top: -2, right: 2, bottom: 2, left: -2 });
+            }, prefix + 'NegativeWidthHeight');
+            test(function() {
+                checkDOMRect(new constructor(0, 0, undefined, 4),
+                    { x: 0, y: 0, width: 0, height: 4, top: 0, right: 0, bottom: 4, left: 0 });
+            }, prefix + 'Undefined1');
```

Comment:
Avoid numbers in test names. Describe what the test is doing in the test name instead.